examples/pipelines/guestbook.yaml

```diff
@@ -8,76 +8,99 @@ spec:
+               - build-push
           params:
               - name: pathToFiles
-                value: guestbook/all-in-one/guestbook-all-in-one.yaml
+                type: guestbook/all-in-one/guestbook-all-in-one.yaml
           clusterBindings:
               - clusterName: test
-          prevTasks:
-              -  build-redis4
-              -  build-guestbook
         - name: int-test-osx                    # 3.a Run Integration tests for osx
           taskRef:
               name: integrationTestInDocker
-          sourceBindings:
-              - inputName: workspace
+          inputSourceBindings:
+              - name: workspace
                 sourceKey: guestbook
+                passedConstraint:
+                - deploy-bundle-test
           params:
               - name: dockerBuildFile
-                value: guestbook-int/Dockerfile
-          prevTasks:
-              - deploy-test
+                type: guestbook-int/Dockerfile
         - name: int-test-linux                  # 3.b Run Integration tests for linux
           taskRef:
               name: integration-test-in-docker
-          sourceBindings:
-              - inputName: workspace
+          inputSourceBindings:
+              - name: workspace
                 sourceKey: guestbook
+                passedConstraint:
+                - deploy-bundle-test
           params:
               - name: dockerBuildFile
-                value: guestbook-int/Dockerfile
-          prevTasks:
-              - deploy-test
+                type: guestbook-int/Dockerfile
         - name: deploy-bundle-staging           # 4. Deploy GuestBook and Redis to staging cluster
           taskRef:
               name: deploy-with-kubectl
-          sourceBindings:
-              - inputName: workspace
+          inputSourceBindings:
+              - name: workspace
                 sourceKey: guestbook
+                passedConstraint:
+                - int-test-osx
+                - int-test-linux
           params:
               - name: pathToFiles
-                value: guestbook/all-in-one/guestbook-all-in-one.yaml
+                type: guestbook/all-in-one/guestbook-all-in-one.yaml
           clusterBindings:
               - clusterName: staging
-          prevTasks:
-              -  int-test-osx
-              -  int-test-linux
+    resources:
+    - name: guestbook
+      resourceRef:
+        name: guestbook
+      type: git
+      url: github.com/kubernetes/examples
+      serviceAccount: githubServiceAccount
+      revision: HEAD
+    - name: redis-docker
+      resourceRef:
+        name: redis-docker    
+      type: git
+      url: github.com/GoogleCloudPlatform/redis-docker/blob/master/4/debian9/4.0/Dockerfile
+      serviceAccount: githubServiceAccount
+      revision: HEAD
+    - name: stagingRegistry
+      resourceRef:
+        name: stagingRegistry      
+      type: image
+      url: gcr.io/demo-staging
+      digest: latest
```

Comment:
i'm having a hard time understanding if this resource represents an "image" or if it represents an "image registry". if it was an image, I would think it might look something like this:

```yaml
    - name: stagingImage
      type: image
      url: gcr.io/demo-staging
      imageName: kritis
```

- added `imageName`
- removed `digest` (won't be known until it's built?)
- changed the name from `stagingRegistry` to `stagingImage` since this resource is an image, not just a registry?

Reply:
makes sense, will make these changes, it should be referring to an image